media/js/base/mozilla-fxa-link.js

```diff
@@ -34,15 +41,66 @@ if (typeof window.Mozilla === 'undefined') {
         return href;
     };
 
+    /**
+     * Returns the hostname for a given URL.
+     * @param {String} url.
+     * @returns {String} hostname.
+     */
+    FxaLink.getHostName = function(url) {
+        var matches = url.match(/^https?:\/\/(?:[^/?#]+)(?:[/?#]|$)/i);
+        return matches && matches[0];
+    };
+
+    /**
+     * Intercept event handler for FxA links and forms, lets the browser drive the FxA Flow using
+     * the `showFirefoxAccounts` UITour API. Attaches several UTM parameters from the current page
+     * that will be forwarded to the browser and later on to FxA services.
+     * @param event {Event}
+     * @private
+     */
+    FxaLink.interceptFxANavigation = function(event) {
+        event.preventDefault();
+        var search = new URL(event.target.href).search;
+        var urlParams = new window._SearchParams(search).params;
+        var extraURLParams = {};
+        var allowedChars = /^[\w/.%-]+$/;
+        var knownParams = ['flowId', 'flowBeginTime', 'deviceId',
+            'entrypoint', 'entrypoint_experiment', 'entrypoint_variation',
+            'utm_source', 'utm_campaign', 'utm_content', 'utm_term', 'utm_medium'];
+
+        for (var i = 0; i < knownParams.length; i++) {
+            var known = knownParams[i];
+            // convert the param to snake_case for the UITour API
+            var knownFormat = known.replace(/([A-Z])/g, ' $1');
```

Comment:
This works fine, but I notice when clicking the links that the final URL query string also contains snake_case params: `&flow_id=76df5964e3e15eae43a6ce13c751414d2fe3b7f96d55c4965bd61e296bbb045a&flow_begin_time=1594374070072&device_id=e801638be46f47d49076faca22eeac28`

Assuming that this format is accepted by the FxA server, I'd perhaps favor changing our default params to snake_case everywhere, as opposed to having to alter them here for this specific flow? You should be able to do that pretty easily here: https://github.com/mozilla/bedrock/blob/master/media/js/base/mozilla-fxa-product-button.js#L80-L82

Reply:
FxA accepts both cases: https://github.com/mozilla/fxa/blob/8a8138b20baf069d31e0a3b1968e4529faaaca0c/packages/fxa-content-server/app/scripts/models/flow.js#L51-L52

So that should be fine

Reply:
That's good to know - if we can change our default above to snake_case to avoid the extra conversion again here then that would help to keep things a bit simpler?

Reply:
I have updated the `flowParams` logic for that, it seems like that was the only spot that had to change.